### ISSUE TITLE
Stop bundling files into a single bundle

### ISSUE THREAD

**Issue by n1ru4l** [kamilkisiela/bob#48]
Instead of bundling everything into a single bundle the original file structure within the `src` folder should be kept.

**Comment by ardatan** [comment]
I think people should be able to configure multiple entrypoints instead of using the source directory structure.

**Comment by n1ru4l** [comment]
I agree with @ardatan on configuring custom entry points. Bob should check the following for the exports map:

1. All the exports point to existing files
2. [All the exports are TypeScript module resolution compatible](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing)

In addition, the bootstrap command (I created in https://github.com/kamilkisiela/bob/pull/55) should help people with onboarding to bob and give good defaults.

________

An overview of the current build process:

Today, transpilation from ts to js is achieved by running `tsc`, followed by bundling and copying files via `bob build`.

In a monorepository the following happens:

`tsc` creates 
- `<monorepo-root>/dist/<package-name>/src/*.js` (esm)
- `<monorepo-root>/dist/<package-name>/src/*.d.ts` (type-defs)

`bob build` does:
- resolves the path of the previous files
- creates a single esm and cjs bundle from the `tsc` emitted files at `<monorepo-root>/packages/<package-name>/dist/index.(m).js`
- copies the type-definitions to `<monorepo-root>/packages/<package-name>/dist/**/*.d.ts`
- generates a `package.json` at  `<monorepo-root>/packages/<package-name>/dist/package.json`
